### PR TITLE
Remove bottom safe-area padding when keyboard up

### PR DESCRIPTION
**Pull request by cbracken**
On the iPhone X, raising the keyboard sets a bottom physical inset equal
to the keyboard height, however we previously did not zero out the bottom
physical padding. This was incorrect; when a soft keyboard is present,
it 'consumes' the safe area inset for the home indicator widget,
eliminating the need for the app to handle this at all.

### CODE REVIEW

**Comment by cbracken** [comment]
Fixes #15947

File: shell/platform/darwin/ios/framework/Source/FlutterViewController.mm [hunk 1]
```diff
@@ -688,8 +688,8 @@ - (void)updateViewportMetrics {
 - (CGFloat)statusBarPadding {
   UIScreen* screen = self.view.window.screen;
   CGRect statusFrame = [UIApplication sharedApplication].statusBarFrame;
-  CGRect viewFrame =
-      [self.view convertRect:self.view.bounds toCoordinateSpace:screen.coordinateSpace];
+  CGRect viewFrame = [self.view convertRect:self.view.bounds
```
Comment (GaryQian) [inline shell/platform/darwin/ios/framework/Source/FlutterViewController.mm]: Not clear what the formatting for .mm is, but looks like auto clang formatted?

File: shell/platform/darwin/ios/framework/Source/FlutterViewController.mm [hunk 2]
```diff
@@ -946,8 +959,9 @@ - (BOOL)isAlwaysUse24HourFormat {
   // the "current system locale" or a custom one. On iOS it only applies to the current
   // system locale. Widget implementors must take this into account in order to provide
   // platform-idiomatic behavior in their widgets.
-  NSString* dateFormat =
-      [NSDateFormatter dateFormatFromTemplate:@"j" options:0 locale:[NSLocale currentLocale]];
+  NSString* dateFormat = [NSDateFormatter dateFormatFromTemplate:@"j"
+                                                         options:0
+                                                          locale:[NSLocale currentLocale]];
```
Comment (GaryQian) [inline shell/platform/darwin/ios/framework/Source/FlutterViewController.mm]: Here too

File: shell/platform/darwin/ios/framework/Source/FlutterViewController.mm (hunk 1 above)
Comment (cbracken) [reply shell/platform/darwin/ios/framework/Source/FlutterViewController.mm]: Yep this was auto-formatted by clang.

**Pull request merged**